backend/corpora/dataset_processing/process.py

```diff
@@ -430,21 +411,50 @@ def main():
         )
         update_db(dataset_id, processing_status=status)
 
-    # Process cxg
-    process_cxg(local_filename, dataset_id, os.environ["CELLXGENE_BUCKET"])
 
-    # Process metadata
-    metadata = extract_metadata(local_filename)
-    logger.info(metadata, flush=True)
-    update_db(dataset_id, metadata)
+def main():
+    check_env()
+    dataset_id = os.environ["DATASET_ID"]
+    try:
+        update_db(dataset_id, processing_status=dict(processing_status=ProcessingStatus.PENDING))
+        local_filename = download_from_dropbox_url(
+            dataset_id,
+            os.environ["DROPBOX_URL"],
+            "local.h5ad",
+        )
+    except ProcessingCancelled:
+        dataset = Dataset.get(dataset_id, include_tombstones=True)
+        dataset.dataset_and_asset_deletion()
+        sys.exit(0)
+    except ProcessingFailed as ex:
+        logging.error(ex.status)
+        update_db(dataset_id, processing_status=dict(processing_status=ProcessingStatus.FAILURE))
+        sys.exit(1)
+    else:
+        logger.info("Download complete", flush=True)
 
-    # create artifacts
-    create_artifacts(
-        local_filename,
-        dataset_id,
-        os.environ["ARTIFACT_BUCKET"],
-    )
-    update_db(dataset_id, processing_status=dict(processing_status=ProcessingStatus.SUCCESS))
+    try:
+        validate_h5ad_file(dataset_id, local_filename)
+        process_cxg(local_filename, dataset_id, os.environ["CELLXGENE_BUCKET"])
+
+        # Process metadata
+        metadata = extract_metadata(local_filename)
+        logger.info(metadata, flush=True)
+        update_db(dataset_id, metadata)
+
+        # create artifacts
+        create_artifacts(
+            local_filename,
+            dataset_id,
+            os.environ["ARTIFACT_BUCKET"],
+        )
+        update_db(dataset_id, processing_status=dict(processing_status=ProcessingStatus.SUCCESS))
+    except ProcessingCancelled:
+        dataset = Dataset.get(dataset_id, include_tombstones=True)
+        dataset.dataset_and_asset_deletion()
```

Comment:
there is not database session handling here. Use a db_session_manager().
```suggestion
        with db_session_manager():
            dataset = Dataset.get(dataset_id, include_tombstones=True)
            db_session_manager()
```